projects/graph/src/graph_demo.py

```diff
@@ -14,13 +14,17 @@ def main(num_vertices=8, num_edges=8):
     for num in range(num_vertices):
         # print(graph.add_vertex(str(num)))
         graph.add_vertex(str(num))
-        print(graph.vertices)
-        for _ in range(num_edges):
-            print(graph.vertices)
-            vertices = sample(graph.vertices.keys(), 2)
+        # print(graph.vertices)
+        # for _ in range(num_edges):
+            # print(graph.vertices.keys())
+    # for num in range(num_vertices):
+    #     for _ in range(num_edges):
+    #         print(graph.vertices)
+            # vertices = sample(graph.vertices.keys(), 2)
+            # print(graph.vertices)
             # TODO check if edge already exsists
-            
-            graph.add_edge(vertices[0], vertices[1])
+            # if graph.vertices.values().get_edges() == vertices:
```

Comment:
What are you trying to do? `graph.vertices.values()` should get you something that's like a list, and in it would contain a bunch of vertex objects. Remember: it's a list(_-like thing?_), so it's _not_ going to have the method `get_edges()`. Only the individual vertex objects would have that.